packages/widget_toolkit/lib/src/lib_item_picker/views/item_picker_page_with_dependencies.dart

```diff
@@ -0,0 +1,67 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+import '../../../widget_toolkit.dart';
+import '../../base/models/item_builder.dart';
+import '../di/item_picker_dependencies.dart';
+import 'item_picker_page.dart';
+
+class ItemPickerPageWithDependencies<T extends PickerItemModel>
+    extends StatelessWidget {
+  const ItemPickerPageWithDependencies({
+    super.key,
+    required this.onTap,
+    required this.service,
+    this.selectedItems,
+    this.isMultiSelect,
+    this.isStatic,
+    this.isItemSelectionRequired,
+    this.loadingItemsCount,
+    this.loadingItemHeight,
+    this.title,
+    this.saveButtonText,
+    this.footerBuilder,
+    this.itemBuilder,
+    this.errorBuilder,
+    this.emptyBuilder,
+    this.separatorBuilder,
+  });
+
+  final List<T>? selectedItems;
+  final Function(List<T?>) onTap;
+  final ItemPickerService<T> service;
+  final bool? isMultiSelect;
+  final bool? isStatic;
+  final bool? isItemSelectionRequired;
+  final int? loadingItemsCount;
+  final double? loadingItemHeight;
+  final String? title;
+  final String? saveButtonText;
+  final WidgetBuilder? footerBuilder;
+  final ItemPickerItemBuilder<T>? itemBuilder;
+  final Widget Function(Exception)? errorBuilder;
+  final Widget Function()? emptyBuilder;
+  final Widget Function(BuildContext context, int index)? separatorBuilder;
+
+  @override
+  Widget build(BuildContext context) => MultiProvider(
```

Comment:
@nikolay-vasilev-prime here is one example where`isInTestMode` should be integrated to enable the golden tests capabilities